extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -0,0 +1,328 @@
+<?php
+
+/**
+ * Main class for the AppPromoLanding screen.  This comandeers /wiki/Community_App and displays
+ * a page with very-little chrome on it that will let a user send the app link to their phone.
+ *
+ * @author Sean Colombo
+ */
+class AppPromoLandingController extends WikiaController {
+	
+	private static $extraBodyClasses = []; // TODO: REMOVE - ONLY DURING THE TEMPORARY COPYING OF OASISCONTROLLER
+
+	const RESPONSE_OK = 200;
+	const APP_CONFIG_SERVICE_URL = "http://prod.deploypanel.service.sjc.consul/api/app-configuration/";
+	const BRANCH_API_URL = "https://api.branch.io/v1/app/";
+
+	// Settings for the background image-grid.
+	const MAX_TRENDING_ARTICLES = 40; // we need about 33 images, and not all articles have images.
+	const THUMBS_PER_ROW = 11;
+	const THUMBS_NUM_ROWS = 3;
+	const IMG_HEIGHT = 184; // sizes directly from Zeplin.io mockup.
+	const IMG_WIDTH = 184;
+
+	protected static $CACHE_KEY = "mobileAppConfigs";
+	protected static $CACHE_KEY_BRANCH = "branchioBranchKey";
+	protected static $CACHE_KEY_VERSION = 0.1;
+	protected static $CACHE_KEY_VERSION_BRANCH = 0.1;
+	protected static $CACHE_EXPIRY = 86400;
+
+	/**
+	 * Render HTML for whole App Promo Landing page.
+	 */
+	public function executeIndex() {
+		wfProfileIn(__METHOD__);
+
+		// Since this "Community_App" article won't be found, we need to manually say it's okay so that it's not a 404.
+		$this->response->setCode(self::RESPONSE_OK);
```

Comment:
There should be spaces inside parentheses, here and throughout the PR. `$this->response->setCode( self::RESPONSE_OK );`